api/test/test_sample.py

```diff
@@ -54,3 +70,51 @@ def test_get_sample(self):
 
         self.assertEqual(get_response.status_code, 200)
         self.assertEqual(list(get_response.data.keys()), self.sample_keys)
+
+    def test_single_gene_query(self):
+        client = APIClient()
+
+        get_response = client.get('/samples?disease=' + self.disease1.acronym +
+                                  '&mutations__gene=' +
+                                  str(self.gene2.entrez_gene_id))
+
+        self.assertEqual(get_response.status_code, 200)
+        self.assertEqual(len(get_response.data['results']), 1)
+        expected_samples = set([self.sample3.sample_id,])
+        returned_samples = set([x['sample_id'] for x in get_response.data['results']])
```

Comment:
Why are you casting these to sets here? I would think that a response returning the same sample multiple times would be a bug.

Reply:
Ahh - it would. I didn't want to deal with order in lists. But I can just sort them. 👍